main/zoe/api/zoe-contract-facet.md

```diff
@@ -166,23 +164,36 @@ Zoe.
 const { issuerKeywordRecord, keywords, terms } = zoe.getInstanceRecord()
 ```
 
-## zcf.getCurrentAllocation(offerHandle, sparseKeywords)
+## zcf.getCurrentAllocation(offerHandle, brandKeywordRecord)
 - `offerHandle` <router-link to="/glossary/#handle">`{Array <Handle>}`</router-link>
-- `sparseKeywords` sparseKeywords is an array of string keywords, which may be a subset of allKeywords.
-- Returns: <router-link to="/zoe/api/records.html#amount-keyword-record">`{<AmountKeywordRecord>}`</router-link>
+- `brandKeywordRecord` <router-link to="/zoe/api/records.html#brand-keyword-record">
+`{[<BrandKeywordRecord>]}`</router-link> is an optional record mapping keywords to brands for each of the offers.
+- Returns: <router-link to="/zoe/api/records.html#amount-keyword-record">
+`{<AmountKeywordRecord>}`</router-link>
+
+For the keywords specified by `brandKeywordRecord, get the amounts currently allocated to the offer. If
+`brandKeywordRecord` is not specified, amounts are returned for only the Keywords that currently have
+allocated amounts (including empty amounts.) This might mean amounts that will be included in the
+payout could be omitted if not specified.
 
-Get the amounts associated with the sparseKeywords for the offer.
+If `brandKeywordRecord` is specified, for keywords without amounts assigned to the offer, empty amounts are filled in 
 
 ```js
 const { foo, bar } = zoe.getCurrentAllocation(offerHandle, ['foo', 'bar']);
 ```
 
-## zcf.getCurrentAllocations(offerHandles, sparseKeywords)
+## zcf.getCurrentAllocations(offerHandles, brandKeywordRecords)
 - `offerHandles` <router-link to="/glossary/#handle">`{Array <Handle>}`</router-link>
-- `sparseKeywords` sparseKeywords is an array of string keywords, which may be a subset of allKeywords.
+- `brandKeywordRecords` <router-link to="/zoe/api/records.html#brand-keyword-record">
+`{[ Array <BrandKeywordRecords>]}`</router-link> `brandKeywordRecords` is an optional array of records
+mapping keywords to brands for each offer.
 - Returns: <router-link to="/zoe/api/records.html#amount-keyword-record">`{[<AmountKeywordRecord>]}`</router-link>
 
-Get a list of the amounts associated with the sparseKeywords for the offers.
+For each `brandKeywordRecord specified keyword, from each offer get its currently allocated amount. If
```

Comment:
The way that I would explain it is, `brandKeywordRecord` is optional, and if it exists, it is used as a filter on the allocation that is returned. Keywords not in `brandKeywordRecord` are dropped, and if the allocation doesn't have a keyword in `brandKeywordRecord`, it is added along with an empty amount. Essentially, the keywords in the allocation returned will be the same as in `brandKeywordRecord` due to dropping some and adding others.